src/frameworks/uikit/ui_application.rs

```diff
@@ -182,15 +182,19 @@ pub(super) fn UIApplicationMain(
         let delegate: id = msg![env; ui_application delegate];
         // IOS 3+ apps usually use application:didFinishLaunchingWithOptions:, and it
         // seems to be prioritized over applicationDidFinishLaunching:.
-        if let Some(launch_with_options) = env.objc.lookup_selector("application:didFinishLaunchingWithOptions:") {
-            if msg![env; delegate respondToSelector:launch_with_options] {
+        match env
+            .objc
+            .lookup_selector("application:didFinishLaunchingWithOptions:")
+        {
+            Some(launch_with_options) if msg![env; delegate respondsToSelector:launch_with_options] =>
```

Comment:
There's going to be many more optional methods like this. Could you extract this into a helper function? The signature would be something like:

```rust
fn responds_to_selector(env: &mut Environment, selector: &str);
```

It could go in `src/objc/objects.rs` perhaps. I'm suggesting a function rather than a method because it'd needs to take a reference to `env` and is quite high-level. Please include a documentation comment about what it does and why (which would be obvious to you I guess).

Reply:
Split this out, also added instance_responds_to_selector. Didn't use the same implementation as in NSObject because I'd like not to rely on objects subclassing NSObject.